mongo/client.go

```diff
@@ -199,7 +200,8 @@ func readPreferenceFromConnString(cs *connstring.ConnString) (*readpref.ReadPref
 			}
 		}
 	}
-	return rp, nil
+
+	return readPref.Primary();
```

Comment:
Let's continue to return `rp` here, which will send `nil` if there is no read preference in the connection string, which is the behavior that would be implied from the function name.